packages/itwin/tree-widget/src/types.ts

```diff
@@ -6,21 +6,21 @@
 import type { SpatialContainmentTreeProps, StagePanelLocation, StagePanelSection } from "@itwin/appui-react";
 import type { SelectableContentDefinition } from "@itwin/components-react";
 import type { IModelConnection, ScreenViewport } from "@itwin/core-frontend";
-import { IPresentationTreeDataProvider } from "@itwin/presentation-components";
 import type { CategoryTreeProps } from "./components/trees/category-tree/CategoriesTree";
 import type { ModelsTreeProps } from "./components/trees/models-tree/ModelsTree";
+import type { Category, ModelInfo } from "./tree-widget-react";
 
 export interface TreeHeaderButtonProps {
-  viewport?: ScreenViewport;
-  iModel?: IModelConnection;
+  viewport: ScreenViewport;
 }
 
 export interface ModelsTreeHeaderButtonProps extends TreeHeaderButtonProps {
-  availableModels?: string[];
+  models?: ModelInfo[];
 }
 
 export interface CategoriesTreeHeaderButtonProps extends TreeHeaderButtonProps {
-  filteredProvider?: IPresentationTreeDataProvider;
+  categories: Category[];
+  filteredCategories?: Category[];
```

Comment:
Since we're making the `Category` type public, I think it needs some renaming:
- type to `CategoryInfo` to match what we have for models
- `key` attribute to `id` or `categoryId`
- `children` attribute to `subCategoryIds`

Reply:
renamed